cvmfs/download.cc

```diff
@@ -231,9 +231,13 @@ static size_t CallbackCurlHeader(void *ptr, size_t size, size_t nmemb,
   if ((info->destination == kDestinationMem) &&
       HasPrefix(header_line, "CONTENT-LENGTH:", true))
   {
-    char *tmp = reinterpret_cast<char *>(alloca(num_bytes+1));
     uint64_t length = 0;
-    sscanf(header_line.c_str(), "%s %"PRIu64, tmp, &length);
+    size_t pos = header_line.find(":") + 1;
+    assert(header_line.length() > pos);
```

Comment:
As mentioned in the [libcurl documentation](https://curl.haxx.se/libcurl/c/CURLOPT_HEADERFUNCTION.html), returning zero will indicate an error to libcurl.  So yes, that's what we should do if the header can't be parsed.